src/cls/IPM/ResourceProcessor/CPF.cls

```diff
@@ -0,0 +1,51 @@
+Class %IPM.ResourceProcessor.CPF Extends %IPM.ResourceProcessor.Abstract
+{
+
+/// Comma-separated list of resource attribute names that this processor uses
+Parameter ATTRIBUTES As STRING = "Path";
+
+/// Description of resource processor class (shown in UI)
+Parameter DESCRIPTION As STRING = "Merges the specified CPF file in the ""Initialize"" lifecycle phase.";
+
+Property Path As %String(MAXLEN = "");
+
+/// Called as phase <var>pPhase</var> is executed for the resource. If <var>pResourceHandled</var> is set to true,
+/// then the default behavior for that resource will be bypassed in the current phase.
+/// Currently, this is only used in the Verify phase, because of different handling of intermediate error statuses.
+/// TODO: Implement for standard database resources (.INC, .CLS, etc.)
+Method OnPhase(pPhase As %String, ByRef pParams, Output pResourceHandled As %Boolean = 0) As %Status
+{
+    Set verbose = $Get(pParams("Verbose"))
+    Set status = $$$OK
+    Try {
+        If pPhase '= "Initialize" {
+            Quit
+        }
+
+        If '$Data(pParams("RootDirectory"), root) {
+            $$$ThrowStatus($$$ERROR($$$GeneralError, "RootDirectory unspecified"))
+        }
+        // Use Construct first, rather than NormalizeFilename, so we don't have to deal with leading/trailing slashes
+        Set filename = ##class(%File).Construct(root, ..Path)  
```

Comment:
@isc-shuliu  I would recommend perhaps having all cpf files under a top level `/cpf` folder by default (similar to how classes under /cls, inc files under /inc etc.). Then `Directory` can optionally override it if needed. The resource name should be the file name. 
cc @isc-eneil @isc-jlechtne @isc-jili @isc-tleavitt thoughts?